static/js/settings_users.js

```diff
@@ -641,17 +641,22 @@ function handle_bot_form($tbody, $status_field) {
             user_id,
             email: bot.email,
             full_name: bot.full_name,
+            user_role_values: settings_config.user_role_values,
+            disable_role_dropdown: bot.is_owner && !page_params.is_owner,
             bot_type: settings_bots.type_id_to_string(bot.bot_type),
         });
 
         let owner_widget;
 
         function submit_bot_details() {
+            const role = Number.parseInt($("#bot-role-select").val().trim(), 10);
             const $full_name = $("#dialog_widget_modal").find("input[name='full_name']");
 
             const url = "/json/bots/" + encodeURIComponent(user_id);
+            const url_for_role_change = "/json/users/" + encodeURIComponent(user_id);
             const data = {
                 full_name: $full_name.val(),
+                role: JSON.stringify(role),
```

Comment:
We should pass `role` field only to the `/json/users` url if we have decided to use this url only not to `/json/bots` url.